homeassistant/components/tado/climate.py

```diff
@@ -151,47 +129,61 @@ class TadoClimate(ClimateDevice):
 
     def __init__(
         self,
-        store,
+        tado,
         zone_name,
         zone_id,
-        data_id,
+        zone_type,
         min_temp,
         max_temp,
         step,
-        ac_device,
-        hot_water_device,
         ac_support_heat,
-        tolerance=0.3,
+        fallback,
     ):
         """Initialize of Tado climate device."""
-        self._store = store
-        self._data_id = data_id
+        self._tado = tado
 
         self.zone_name = zone_name
         self.zone_id = zone_id
+        self.zone_type = zone_type
 
-        self._ac_device = ac_device
-        self._hot_water_device = hot_water_device
+        self._ac_device = zone_type == TYPE_AIR_CONDITIONING
         self._ac_support_heat = ac_support_heat
         self._cooling = False
 
         self._active = False
         self._device_is_active = False
 
-        self._unit = TEMP_CELSIUS
         self._cur_temp = None
         self._cur_humidity = None
         self._is_away = False
         self._min_temp = min_temp
         self._max_temp = max_temp
         self._step = step
         self._target_temp = None
-        self._tolerance = tolerance
+
+        if fallback:
+            _LOGGER.debug("Default overlay is configured in TADO MODE")
```

Comment:
Please move side effects out of init method.

Reply:
Hmm, this doesn't do any IO, it just sets the default_overlay variable. I've modified the debug statement a bit.